sdk/src/utils/xmp_inmemory_utils.rs

```diff
@@ -90,19 +90,38 @@ fn extract_xmp_key(xmp: &str, key: &str) -> Option<String> {
 }
 
 // writes the event to the writer)
-/// Add a value to XMP using a key, replaces the value if the key exists
+/// Add a value to XMP using a key, replaces the value if the key exists, or adds it only once
 fn add_xmp_key(xmp: &str, key: &str, value: &str) -> Result<String> {
-    let mut reader = Reader::from_str(xmp);
-    reader.config_mut().trim_text(true);
-    let mut writer = Writer::new_with_indent(Cursor::new(Vec::new()), b' ', 2);
+    let orig_length = xmp.len();
+
+    // Minimal padding should be 2 KB to 4 KB. This is used if no XMP packet is found.
+    let mut target_length = 4096;
+    // Remove trailing xpacket if present for easier manipulation
+    let xpacket_end_single = "<?xpacket end='w'?>";
```

Comment:
Single quotes are what the spec has, but we specifically test for double quotes. The test images from cameras also use double quotes.
`<?xpacket end='w'?>`

Reply:
I think this needs to be more flexible.  You may also encounter end='r'.

Reply:
Or we could be a good citizen and respect the 'r' which means read only.